### ISSUE TITLE
How to handle symbol size?

### ISSUE THREAD

**Issue by pelord** [bojko108/ol-esri-style#26]
https://github.com/bojko108/ol-esri-style/blob/118d7cf33ac43ec587362ff7aca56e6ca371ed8c/src/index.js#L276

![image](https://user-images.githubusercontent.com/7397743/173108482-673a85ff-2d14-4675-aa7b-53ce1b23c7ed.png)

vs 

![image](https://user-images.githubusercontent.com/7397743/173108913-6df530cf-4be3-4ac8-9846-e4b5bbbef532.png)

**Comment by bojko108** [comment]
Hello,
as I see only `imageData` and `rotation` parameters are taken from the symbol definition provided by the ArcGIS Server response. Do you know the actual size of the image you use? Is it possible for you to test with a smaller icon? I think OpenLayers take the actual image size and displays your icon with it, so using a smaller image will probably help in this case.

One more thing: is it possible to provide me with the JSON response from the ArcGIS Server? I can try to extend the code and read more parameters from it instead only the symbol rotation.

**Comment by pelord** [comment]
I dont have any control on the provided server.. ;( I don't know the original size (to process scale).

Here the json response:

https://services5.arcgis.com/52xIcscE2n4IFiR7/arcgis/rest/services/R%C3%A9pertoire_FQK/FeatureServer/1?f=pjson

![image](https://user-images.githubusercontent.com/7397743/173112250-5cf8d42d-5a5e-4111-a4a7-5a1ec9a2c86f.png)

**Comment by bojko108** [comment]
Thank you for this information. For sure it will be better to extend this module to read more parameters from the JSON response. I'm a little busy right now, but will try to make a PR tomorrow. 

Something like this:

```js
case 'esriPMS':
    return {
      icon: {
        src: `data:image/png;base64,${symbol.imageData}`,
        rotation: symbol.angle,
        size: [symbol.width, symbol.height]    // units should be checked, ol expects pixels 
      },
    };

**Comment by pelord** [comment]
Thanks for the quick reply. 

I've tried with the size property.. but it is simply "crop" the image instead of scaling it...

**Comment by bojko108** [comment]
Hi,
you are right, it seems that `size` parameter sets what part of the image you want to use instead of the absolute size of the image on the map (documented [here](https://openlayers.org/en/latest/apidoc/module-ol_style_Icon-Icon.html)). I tried to use `scale` instead and it works. You can check it here: [https://codesandbox.io/s/modify-icon-forked-sottly?file=/main.js](https://codesandbox.io/s/modify-icon-forked-sottly?file=/main.js)

```js
image: new Icon({
    // size: [20, 20], // do not specify the size parameter, use the whole picture instead
    scale: 0.2,
    src: 'data:image/png;base64,iVBORw0KGgoAAAANSUhE...
})
```

Now the question is how to modify the code of `ol-esri-style` :). Should we set the size parameter based on the `width` and `height` from ArcGIS Server or do we calculate the `scale` somehow?

**Comment by bojko108** [comment]
You could also modify the styles like this (it's ugly but can help):
```js
createStyleFunctionFromUrl(layerDefinition.url).then((styleFunction) => {
    vector.setStyle((feature, resolution) => {
        const styles = styleFunction(feature, resolution);
        return styles.map((style) => {
            const image = style.getImage();
            if (image instanceof Icon) {
                // modify the styles in place
                image.size_ = undefined;
                image.setScale(0.2);
            }
            return style;
        });
    });
});
```

**Comment by pelord** [comment]
do you think something like that would work? 

https://stackoverflow.com/questions/60571105/getting-value-of-image-width-from-javascript-image-onload-function

**Comment by pelord** [comment]
seems to be an isssue on ol since a long time. 
https://github.com/openlayers/openlayers/issues/13508

**Comment by pelord** [comment]
I've found a fix... 

https://codesandbox.io/s/modify-icon-forked-ez2zos

I've simply follow the @mike-000  instructions: https://github.com/openlayers/openlayers/issues/13508#issuecomment-1079686598

**Comment by bojko108** [comment]
Yes, it is simmilar to what I'm trying. I'll publish the code tomorrow. The idea is that the image is loaded and then we calculate the scale based on the actual image size and the target size on top of the map. The problem is that there is no synchronious way to get the image size so I have to make everything async.

**Comment by bojko108** [comment]
You can check the latest version - v2.1.0, to see if the issue is resolved.

**Comment by pelord** [comment]
I will give a try soon! Thanks!
